File: zhiqian/web/src/api/project.ts
import http from './http'

export interface Project {
  id: number
  name: string
  sourceDb: string
  targetDb: string
  framework: string
  description: string
  status: string
  createdAt: string
}

export function listProjects(): Promise<Project[]> {
  return http.get('/projects')
}

export function getProject(id: number): Promise<Project> {
  return http.get(`/projects/${id}`)
}

File: zhiqian/backend/src/main/java/com/zhiqian/project/Project.java
package com.zhiqian.project;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.Instant;

@Data
@Table("project")
public class Project {
    @Id
    private Long id;
    private String name;
    @Column("source_db")
    private String sourceDb;
    @Column("target_db")
    private String targetDb;
    private String framework;
    private String description;
    private String status;
    @Column("created_at")
    private Instant createdAt;
}
